module/builder/consensus/builder.go

```diff
@@ -283,10 +315,19 @@ func (b *Builder) BuildOn(parentID flow.Identifier, setter func(*flow.Header) er
 			break
 		}
 
-		seals = append(seals, next)
+		nextErToBeSealed := next.IncorporatedResult.Result
+		if len(nextErToBeSealed.Chunks) < 1 {
+			return nil, fmt.Errorf("ExecutionResult without chunks: %v", nextErToBeSealed.ID())
+		}
+		initialState := nextErToBeSealed.Chunks[0].StartState
```

Comment:
Can we add a `InitialStateCommit` method to `ExecutionResult` instead?

Reply:
🎵